examples/with-stomp/pages/_app.js

```diff
@@ -0,0 +1,39 @@
+import App from 'next/app'
+import React from 'react'
+import { Stomp } from '@stomp/stompjs'
+
+class MyApp extends App {
+  static async getInitialProps ({ Component, ctx }) {
+    let pageProps = {}
+
+    if (Component.getInitialProps) {
+      pageProps = await Component.getInitialProps(ctx)
+    }
+
+    return { pageProps }
+  }
+
+  state = {
+    stompClient: null
+  }
+
+  componentDidMount () {
+    // connect to Websocket only once.
+    const client = Stomp.client('ws://some.stomp.server')
```

Comment:
Why you don't ensure in the `withSTOMP` hook that you work with a singleton of the client? In that way you don't need a custom `_app` and don't disable [prerendering optimization](https://github.com/zeit/next.js/#automatic-prerendering).

Reply:
@lfades @StarpTech That sounds great, I'll do it. Thank you.